docs/source/classes.rst

```diff
@@ -616,3 +616,100 @@ Methods
 
 .. automethod:: dallinger.models.Question.fail
 
+
+Database Session Management
+---------------------------
+
+Database sessions are handled automatically by Dallinger in the following cases:
```

Comment:
I think we should begin this section by explaining what a database session is and how this is capture in Dallinger. I think we also need to contrast session creation with transaction committing, and help readers to understand when they need to call `commit()` themselves (`scheduled_task` seems to be an odd one out where the reader needs to call it manually).